tests/codeception/common/_support/FixtureHelper.php

```diff
@@ -58,6 +58,30 @@ public function globalFixtures()
     }
 
     /**
+     * Workaround used to load custom fixtures for the test
+     * @link https://github.com/Codeception/Codeception/issues/2258
+     * @link https://github.com/yiisoft/yii2/issues/5575
+     * @param array $fixtureConfig
+     */
+    public function loadFixturesForTest(array $fixtureConfig)
+    {
+        $fixtures = ArrayHelper::merge(
```

Comment:
> $this->loadFixtures() will truncate all tables which has a relevant fixture. So if you have a global fixture for user table you have to explicitly load it in $fixtureConfig or all this data will be overridden by $this->fixtures() 

this is correct behavior, in global fixtures as was said you load global data it can be read - only or not, but it is developer's right to do so. He should be aware of this situation. That is why i recommend to split it as was said. The only thing i recommended is to add `getFixture` as public and make `loadFixtures($fixtures)` as public, this should be enough in most cases.
The point is:

- load global fixtures / unload in before / after suite;
- in your test load / unload other fixtures if needed

thats it, you should not join them together in one file as you done.